internal/pkg/service/stream/aggregation/repository/repository.go

```diff
@@ -0,0 +1,198 @@
+	statsRepo "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics/repository"
+)
+
+type dependencies interface {
+	Logger() log.Logger
+	EtcdClient() *etcd.Client
+	DefinitionRepository() *definitionRepo.Repository
+	StatisticsRepository() *statsRepo.Repository
+	StorageRepository() *storageRepo.Repository
+}
+
+type Repository struct {
+	logger     log.Logger
+	client     etcd.KV
+	definition *definitionRepo.Repository
+	statistics *statsRepo.Repository
+	storage    *storageRepo.Repository
+}
+
+type SourceWithSinks struct {
+	SourceKey key.SourceKey
+	Sinks     []*SinkWithStatistics
+}
+
+type SinkWithStatistics struct {
+	*definition.Sink
+	Statistics *SinkStatistics
+}
+
+type SinkStatistics struct {
+	Total *statistics.Aggregated
+	Files []*FileWithStatistics
+}
+
+type FileWithStatistics struct {
+	*model.File
+	Statistics *statistics.Aggregated
+}
+
+func New(d dependencies) *Repository {
+	return &Repository{
+		logger:     d.Logger().WithComponent("aggregation.repository"),
+		client:     d.EtcdClient(),
+		definition: d.DefinitionRepository(),
+		statistics: d.StatisticsRepository(),
+		storage:    d.StorageRepository(),
+	}
+}
+
+func (r *Repository) GetSourcesWithSinksAndStatistics(ctx context.Context, sourceKeys []key.SourceKey) ([]*SourceWithSinks, error) {
+	res, err := r.findSinksForSources(ctx, sourceKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.addStatisticsToAggregationResponse(ctx, res)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.addFilesToAggregationResponse(ctx, res)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.addFileStatisticsToAggregationResponse(ctx, res)
+	if err != nil {
+		return nil, err
+	}
+
+	return maps.Values(res), nil
+}
+
+func (r *Repository) findSinksForSources(ctx context.Context, sourceKeys []key.SourceKey) (map[key.SourceKey]*SourceWithSinks, error) {
+	res := make(map[key.SourceKey]*SourceWithSinks)
+
+	txn := op.Txn(r.client)
+	for _, sourceKey := range sourceKeys {
+		txn.Merge(
+			r.definition.Sink().List(sourceKey).ForEach(func(value definition.Sink, header *iterator.Header) error {
+				source, ok := res[value.SourceKey]
+				if !ok {
+					source = &SourceWithSinks{
+						SourceKey: value.SourceKey,
+					}
+					res[value.SourceKey] = source
+				}
+				source.Sinks = append(source.Sinks, &SinkWithStatistics{Sink: &value})
+				return nil
+			}),
+		)
+	}
+
+	err := txn.Do(ctx).Err()
+
+	return res, err
+}
+
+func (r *Repository) addStatisticsToAggregationResponse(ctx context.Context, res map[key.SourceKey]*SourceWithSinks) error {
+	txn := op.Txn(r.client)
+	for sourceKey, source := range res {
+		for _, sink := range source.Sinks {
+			sinkKey := key.SinkKey{
+				SourceKey: sourceKey,
+				SinkID:    sink.SinkID,
+			}
+
+			txn.Merge(r.statistics.AggregateIn(sinkKey).OnResult(func(result *op.TxnResult[statistics.Aggregated]) {
+				sink.Statistics = &SinkStatistics{
+					Total: ptr.Ptr(result.Result()),
+				}
+			}))
```

Comment:
Cannot we do double `Merge` to prevent having 2 functions doing very simillar loop? It seems to be unoptimized
```
txn.Merge(r.statistics.AggregateIn(sinkKey).OnResult(func(result *op.TxnResult[statistics.Aggregated]) {
sink.Statistics = &SinkStatistics{
	Total: ptr.Ptr(result.Result()),
}
}))
txn.Merge(r.storage.File().ListRecentIn(sinkKey).ForEach(func(value model.File, header *iterator.Header) error {
sink.Statistics.Files = append(sink.Statistics.Files, &FileWithStatistics{
	File: ptr.Ptr(value),
})
return nil
}))
```